homeassistant/components/automation/__init__.py

```diff
@@ -313,16 +318,37 @@ def as_dict(self) -> Dict[str, Any]:
         return result
 
 
+class LimitedSizeDict(OrderedDict):
+    """OrderedDict limited in size."""
+
+    def __init__(self, *args, **kwds):
+        """Initialize OrderedDict limited in size."""
+        self.size_limit = kwds.pop("size_limit", None)
+        OrderedDict.__init__(self, *args, **kwds)
+        self._check_size_limit()
+
+    def __setitem__(self, key, value):
+        """Set item and check dict size."""
+        OrderedDict.__setitem__(self, key, value)
+        self._check_size_limit()
+
+    def _check_size_limit(self):
+        """Check dict size and evict the oldest item if needed."""
```

Comment:
If a value is updated, it's key will now still be considered `oldest`. 

I guess in this case it doesn't matter because every key is only set once.

Reply:
Comment updated in c7dd806